scipy/stats/_sobol.pyx

```diff
@@ -0,0 +1,297 @@
+from __future__ import division, absolute_import
+
+cimport cython
+cimport numpy as cnp
+
+import os
+import math
+import numpy as np
+from scipy.stats import norm
+
+cdef int MAXDIM = 21201  # max number of dimensions
+cdef int MAXDEG = 18  # max polynomial degree
+cdef int MAXBIT = 30  # max number of bits
+
+cdef int poly[21201]
+cdef int vinit[21201][18]
+
+cdef int LARGEST_NUMBER = 2 ** MAXBIT  # largest possible integer
+cdef float RECIPD = 1.0 / LARGEST_NUMBER  # normalization constant
+
+cdef bint is_initialized = False
+
+
+# Load direction numbers (taken from https://web.maths.unsw.edu.au/~fkuo/sobol/)
+def initialize_direction_numbers():
+    global is_initialized, poly, vinit
+    if not is_initialized:
+        dns = np.load(os.path.join(os.path.dirname(__file__), "_sobol_direction_numbers.npz"))
+        poly = dns["poly"]
+        vinit = dns["vinit"]
+        is_initialized = True
```

Comment:
This file contains numbers which we cannot regenerate at build time. I believe @Balandat just put the relevant information in the file. We could download the file at build time, but I am afraid the data will not be on the website forever. I will see to add a description of the file in this function with what Max already wrote.

Reply:
@Balandat, also here

Reply:
@tupui is right that we cannot do the direction number generation at build time (well we could in theory but it would be big hassle and take a long time). Instead we use the direction numbers from the reference. These are provided in a text file and not in the format that we'd like for the code here. To avoid doing that conversion, I did the conversion once and packaged it up as an .npz file. That simplifies and speeds up things substantially vs. re-loading a 2MB text file. I have code for the extraction+conversion+saving in python. I can either check that in as a .py file that is part of the source (but not part of the package), or I can make an issue with the code paste and link it here. @rgommers, any preferences?

Reply:
Added the code (pretty short) as part of the docstring for `initialize_direction_numbers`.